src/Components/Components/src/Routing/LegacyRouteMatching/LegacyRouteTableFactory.cs

```diff
@@ -18,6 +19,7 @@ internal static class LegacyRouteTableFactory
             new ConcurrentDictionary<Key, LegacyRouteTable>();
         public static readonly IComparer<LegacyRouteEntry> RoutePrecedence = Comparer<LegacyRouteEntry>.Create(RouteComparison);
 
+        [RequiresUnreferencedCode("This API calls Assembly.ExportedTypes. Types and members of the loaded assembly might be removed during trimming.")]
```

Comment:
This will be a public facing warning message. So explaining internal implementation details doesn't help. Ideally it should tell the user what to do about this - call a different api, or something similar.